docs: improve caching documentation (#4081)
tuist/tuist: projects/docs/docs/commands/cache.md

---
title: tuist cache
slug: '/commands/cache'
description: "Learn how to use Tuist's cache command to generate binary artifacts for your targets."
---

Target caching is one of Tuist's distictive features.
Caching creates binary artifacts of your targets, which can later be used in your project if your generation is focused on other targets.

> For more details on the caching workflow, please refer to the [caching]](/building-at-scale/caching/) documentation.

Reviewer: ```suggestion
For more details on the caching workflow, please refer to the [caching]](building-at-scale/caching.md) documentation.
```

> Target caching is one of Tuist's distictive features.
> It creates a binary artifact of your targets, which can later be used in your project if your generation is focused on other targets.
> For more details of the overall caching workflow, you can read the [caching]](/building-at-scale/caching/) documentation.

Reviewer 2: ```suggestion
Target caching is one of Tuist's distictive features.
Caching creates binary artifacts of your targets, which can later be used in your project if your generation is focused on other targets.
For more details on the caching workflow, please refer to the [caching]](/building-at-scale/caching/) documentation.
```

### Warm

To generate the cache for all the target of your project, youn can simply run:

```bash
tuist cache warm
```

You can even select a subset of targets to be cached, by passing them as command arguments:

```bash
tuist cache warm TargetA TargetB
```

> The cached artifacts are stored in Tuist cache at `~/.tuist/Cache/BuildCache/<target-hash>`.
> If a target is already present in cache, it will not be built again by the command.

Reviewer 2: ```suggestion
The cached artifacts are stored in the Tuist cache at `~/.tuist/Cache/BuildCache/<target-hash>`.
If a target is already present in the cache, it will not be built again by the command.
```

### Arguments

| Argument              | Short | Description                                                                                                                    | Default                                         | Required |
| --------------------- | ----- | ------------------------------------------------------------------------------------------------------------------------------ | ----------------------------------------------- | -------- |
| `--path`              | `-p`  | The path to the directory that contains the definition of the project.                                                         | Current directory                               | No       |
| `--profile`           | `-P`  | The name of the profile to be used when warming up the cache.                                                                  |                                                 | No       |
| `--xcframeworks`      | `-x`  | When passed it builds xcframeworks (simulator and device) instead of frameworks (only simulator).                              | False                                           | No       |

> | `--dependencies-only` | `N/A` | If passed, the command doesn't cache the targets passed in the `--targets` argument, but only their dependencies.              | False                                           | No       |

Reviewer 2: A bit misleading, because I don't think there is a `--targets` flag, right?

```suggestion
| `--dependencies-only` | `N/A` | If passed, the command caches only the dependencies of the list of targets passed to the cache command.              | False                                           | No       |
```

Author: nope, it was correct: if you to `tuist cache warm A --dependencies-only`, then A is not cached, but only its dependencies.
Thanks for all the other suggestions!

Reviewer 2: A right, my change still got that point across.

What I'm wondering though is, is `--targets` a valid flag? I wasn't aware that you could use `--targets` in the command. If so, we should document that explicitly here in this table.
